app/controllers/application_controller.rb

```diff
@@ -56,16 +56,35 @@ def rewrite_v4_ids
 
   def store_location
     store_location_for(:user, request.url)
-      if request.env["omniauth.params"].present? && request.env["omniauth.params"]["login_popup"].present?
-        session[:previous_url] = root_path + "self_closing.html"
-      end
+    if request.env["omniauth.params"].present? && request.env["omniauth.params"]["login_popup"].present?
+      session[:previous_url] = root_path + "self_closing.html"
+    end
   end
 
-  def after_sign_in_path_for(resource)
+  def after_sign_in_path_for(_resource)
     if request.env['QUERY_STRING']['login_popup'].present?
       root_path + "self_closing.html"
     else
-      request.env['omniauth.origin'] || stored_location_for(resource) || session[:previous_url] || root_path
+      request.env['omniauth.origin'] || find_redirect_url(nil)
```

Comment:
Without stored_location_for, logging in from embedded player may not auto-close popup window. I think we could merge now and test this use case on Spruce.